index.d.ts

```diff
@@ -1,6 +1,6 @@
 import { IncomingMessage, ServerResponse } from 'http'
 
-export type RequestIdFactory = () => unknown
+export type RequestIdFactory = (req?: IncomingMessage) => unknown
```

Comment:
In fact, the `IncomingMessage` type for `req` will only hold true. E.g. for Fastify it'll be [`FastifyRequest`](https://github.com/fastify/fastify/blob/master/types/request.d.ts). Maybe it's better to use generics here (or `any`, but that's not convenient for TS users).